dev/cassandra-server/src/main/scala/com/lightbend/lagom/internal/cassandra/CassandraLauncher.scala

```diff
@@ -5,183 +5,78 @@
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy
+import org.testcontainers.shaded.org.apache.commons.io.FileUtils
 
 /**
  * Starts Cassandra.
- *
- * Adapted from akka.persistence.cassandra.testkit.CassandraLauncher.
  */
 class CassandraLauncher {
 
-  private val ForcedShutdownTimeout              = 20.seconds
   private var cassandraDaemon: Option[Closeable] = None
 
-  private val devEmbeddedYaml = "dev-embedded-cassandra.yaml"
+  private var server: CassandraContainer[_] = _
+  private var proxy: TcpProxy = _
 
+  var hostname: String = "127.0.0.1"
   var port: Int = 0
 
   def address = s"$hostname:$port"
 
-  def hostname = "127.0.0.1"
-
   def start(
       cassandraDirectory: File,
       yamlConfig: File,
       clean: Boolean,
-      port: Int,
-      jvmOptions: Array[String]
+      port: Optional[Int], // Can't use scala.Option, because this method is called from another ClassLoader.
+      loader: ClassLoader,
+      maxWaiting: java.time.Duration
   ): Unit = this.synchronized {
-
     if (cassandraDaemon.isEmpty) {
-
       prepareCassandraDirectory(cassandraDirectory, clean)
-
-      val storagePort = AkkaCassandraLauncher.freePort()
-      // NOTE: yamlConfig will be null when not explicitly configured by user
-      // we don't use an Option for it because this class will be dynamically loaded
-      // and called using structural typing (reflection) by sbt thus on a classloader with scala 2.10
-      val fileOrResource = Either.cond(yamlConfig == null, devEmbeddedYaml, yamlConfig)
-      // http://wiki.apache.org/cassandra/StorageConfiguration
-      val conf = readResource(fileOrResource)
-      val amendedConf = conf
-        .replace("$PORT", port.toString)
-        .replace("$STORAGE_PORT", storagePort.toString)
-        .replace("$DIR", cassandraDirectory.getAbsolutePath)
-
-      // write yaml file in cassandra dir (target/embedded-cassandra)
-      val devConfigFile = new File(cassandraDirectory, devEmbeddedYaml)
-      writeToFile(devConfigFile, amendedConf)
-
-      // Extract the cassandra bundle to the directory
-      val cassandraBundleFile = new File(cassandraDirectory, "cassandra-bundle.jar")
-      if (!cassandraBundleFile.exists()) {
-        val is =
-          this.getClass.getClassLoader.getResourceAsStream("akka/persistence/cassandra/launcher/cassandra-bundle.jar")
-        try {
-          Files.copy(is, cassandraBundleFile.toPath)
-        } finally {
-          is.close()
+      val currentClassLoader: ClassLoader = Thread.currentThread().getContextClassLoader
+      try {
+        Thread.currentThread().setContextClassLoader(loader)
+        server = new CassandraContainer().withStartupAttempts(1)
+        Thread.currentThread()
+        server = server.withFileSystemBind(cassandraDirectory.getAbsolutePath, "/var/lib/cassandra")
+        if (yamlConfig != null) {
+          server = server.withFileSystemBind(yamlConfig.getAbsolutePath, "/etc/cassandra/cassandra.yaml")
```

Comment:
Is this portable? Are those paths referring to the host machine or the container?

Reply:
> Is this portable?

I run this only on my Mac 😄 

> Are those paths referring to the host machine or the container?

```
* @param hostPath the file system path on the host
* @param containerPath the file system path inside the container
```
The first argument on the host, second - inside the container. It's the same as `-v <host_path>:<container_path>` for `docker run` command.

Reply:
I would recommend using `withCopyFileToContainer` that will copy the config instead of mounting it